promql/promqltest/test.go

```diff
@@ -806,7 +830,7 @@ func validateExpectedAnnotations(expr string, expectedAnnotations []expectCmd, a
 	for _, e := range expectedAnnotations {
 		matchFound := slices.ContainsFunc(actualAnnotations, e.CheckMatch)
 		if !matchFound {
-			return fmt.Errorf("expected %s annotation %q but no matching annotation was found for query %q (line %d)", annotationType, e.String(), expr, line)
+			return fmt.Errorf(`expected %s annotation matching %s %q but no matching annotation was found for query %q (line %d), found: %v`, annotationType, e.Type(), e.String(), expr, line, actualAnnotations)
```

Comment:
`actualAnnotations` here will only include annotations of type `annotationType` - I think it'd be useful to include all annotations (of all types) in case the issue is that the expectation and actual annotation have different types.

Reply:
I'll add.

Reply:
Done!